### PR TITLE
Vector FD Table

### PR DESCRIPTION
**Pull request by rennergade**
This changes the implementation of the fdtable from a DashMap of locked fds to a vector of locked fds. This removes one layer of locking by getting rid of the DashMap, and thus improves I/O performance.

### CODE REVIEW

File: src/safeposix/syscalls/net_calls.rs [hunk 1]
```diff
@@ -44,7 +44,12 @@ impl Cage {
     }
 
     fn _socket_inserter(&self, sockfd: SocketDesc) -> i32 {
-        return self.get_next_fd(None, Socket(sockfd));
+
```
Comment (jesings) [inline src/safeposix/syscalls/net_calls.rs:47]: This space is odd

File: src/safeposix/syscalls/net_calls.rs [hunk 2]
```diff
@@ -523,7 +520,7 @@ impl Cage {
                             };
```
Comment (jesings) [inline src/safeposix/syscalls/net_calls.rs:521]: Again it may be possible to omit this line but I'm not sure

**Review by jesings — APPROVED** [review]
This looks a lot cleaner and overall great!

**Review by TristanB22 — APPROVED** [review]
I agree with what Jonathan noted on this PR. Looking through the implementations that you added, everything seems to make sense to me. Although, I am a little confused why the Dashmap implementation wasn't adding the performance improvements that we were hoping for and now we are being locks back. But, I will defer to you all on that one.

**Comment by rennergade** [comment]
@TristanB22 We're not actually "bringing locks back". The individual fds have always had their own locks. We moved from a globally locked hashmap -> dashmap to remove that outer lock which gave us some speed improvements, but dashmap still has some concurrency needs that makes it slower than what were changing to here. With this version we don't need any outer concurrency primitives at all.

**Pull request merged**